tests/test_recommendation.py

```diff
@@ -229,6 +221,47 @@ def test_find_by_id_relid(self):
 
         self.assertRaises(TypeError, Recommendation.find_by_id_relid, 1, "not_int")
         self.assertRaises(TypeError, Recommendation.find_by_id_relid, "not_int", 1)
+    
+    def test_find_by_rel_id(self):
+        """ Test find by related_product_id"""
+        recommendation = self._create_one_recommendation(1, 2, 1)
+        result = Recommendation.find_by_rel_id(recommendation.related_product_id)
+        self.assertEqual(result.first(), recommendation)
+
+    def test_find_by_relid_status(self):
+        """ Test find by related_product_id and status"""
+        recommendation1 = self._create_one_recommendation(1, 2, 1)
+        recommendation2 = self._create_one_recommendation(2, 2, 3, by_status=False)
+        result = Recommendation.find_by_relid_status(2, False)
+        self.assertEqual(len(result.all()), 1)
+        self.assertEqual(result.first(), recommendation2)
+
+        self.assertRaises(TypeError, Recommendation.find_by_relid_status, 2, 1)
+        self.assertRaises(TypeError, Recommendation.find_by_relid_status, "String", True)
+
+    def test_find_by_relid_type(self):
+        """ Test find by related_product_id and type_id"""
+        recommendation1 = self._create_one_recommendation(1, 2, 3)
+        recommendation2 = self._create_one_recommendation(2, 2, 3, by_status=False)
+        result = Recommendation.find_by_relid_type(2, 3)
+        self.assertEqual(len(result.all()), 2)
+
+        self.assertRaises(TypeError, Recommendation.find_by_relid_type, 1, "I'm everything but not int")
+        self.assertRaises(TypeError, Recommendation.find_by_relid_type, "I'm everything but not int", 1)
+        self.assertRaises(DataValidationError, Recommendation.find_by_relid_type, 5, 999999)
+
+    def test_find_by_relid_type_status(self):
+        """ Test find by related product id with type id and status"""
+        recommendation1 = self._create_one_recommendation(1, 2, 3)
+        recommendation2 = self._create_one_recommendation(2, 2, 3, by_status=False)
```

Comment:
Minor issue, but should we create recommendation record with two ids being the same (line 256 in `test_recommendation.py`? (Maybe we should put constraint in `_create_one_recommendation`?)

Reply:
Ah, you are right. I didn't notice this. Thanks for pointing out. Now the function works fine for the test but putting a constraint in `_create_one_recommendation` may be a good idea.